hypershift-operator/controllers/uwmtelemetry/uwm_telemetry.go

```diff
@@ -0,0 +1,323 @@
+package uwmtelemetry
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/go-logr/logr"
+	monv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/openshift/hypershift/hypershift-operator/controllers/manifests/monitoring"
+	"github.com/openshift/hypershift/support/upsert"
+)
+
+const (
+	telemetryRemoteWriteURL = "https://infogw.api.openshift.com/metrics/v1/receive"
+)
+
+type UWMTelemetryReconciler struct {
+	client.Client
+	upsert.CreateOrUpdateProvider
+}
+
+func (r *UWMTelemetryReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	_, err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Namespace{}).
```

Comment:
Hmm, it should. Need to figure out how to limit the configmaps that it can watch

Reply:
You can use a predicate. It might make sense to write a small test that runs a `Reconcile`, stores which objects where updated and then verifies that that set is identical to the set of objects the predicate allows